### PR TITLE
83862 jobs in web console

### PR DESCRIPTION
**Pull request by satr**

### CODE REVIEW

File: api/deployments/deployment_controller.go [hunk 1]
```diff
@@ -40,6 +40,11 @@ func (dc *deploymentController) GetRoutes() models.Routes {
 			Method:      "GET",
 			HandlerFunc: GetPodLog,
 		},
+		models.Route{
+			Path:        rootPath + "/deployments/{deploymentName}/jobcomponents/{jobComponentName}/scheduledjobs/{scheduledJobName}/logs",
```
Comment (nilsgstrabo) [inline api/deployments/deployment_controller.go:44]: I think we should use a better Path for logs, perhaps `/applications/{appName}/environments/{envName}/jobcomponents{jobComponentName}/schedulerjobs/{schedulerJobName}/logs`
The handler for this route has a TODO, copied from logs for replicas, and I don't think we should introduce TODOs in new code when we can do it correct. There is no backward compatibility that we need to consider, like for the existing logs endpoint for replicas.
`deploymentName` and `jobComponentName` from the route is not used in the controller or handler code, and the handler reads the `RadixApplication` and enumerates all enviornments in this spec to get enviroment namespace and queries this namespace for the existence of the pod. A lot of extra API calls to the k8s API to check if a pod exist. By using the the suggestet route Path we can extract the environment name from the route. No need to enumerate environments in RadixApplication . Also, we have the `jobComponentName` and can check that the pod defined by `schedulerJobName` actually belongs to the component.

File: api/deployments/get_deployment_handler.go [hunk 2]
```diff
@@ -57,7 +57,29 @@ func (deploy DeployHandler) GetLogs(appName, podName string, sinceTime *time.Tim
 	return "", deploymentModels.NonExistingPod(appName, podName)
 }
 
-// GetDeploymentsForApplication Lists deployments accross environments
+// GetScheduledJobLogs handler for GetScheduledJobLogs
+func (deploy DeployHandler) GetScheduledJobLogs(appName, scheduledJobName string, sinceTime *time.Time) (string, error) {
+	ns := crdUtils.GetAppNamespace(appName)
+	// TODO! rewrite to use deploymentId to find pod (rd.Env -> namespace -> pod)
```
Comment (nilsgstrabo) [inline api/deployments/get_deployment_handler.go:63]: See comments in the controller.

File: docs/docs.go [hunk 3]
```diff
@@ -4,9 +4,9 @@
 //
 //     Schemes: http, https
 //     BasePath: /api/v1
-//     Version: 1.7.1
+//     Version: 1.9.0
 //     Contact: https://equinor.slack.com/messages/CBKM6N2JY
-//
+//api/deployments/component_handler.go
```
Comment (nilsgstrabo) [inline docs/docs.go:9]: What is this?

**Review by nilsgstrabo — CHANGES_REQUESTED** [review]
A few comments

File: api/deployments/component_handler.go [hunk 4]
```diff
@@ -129,46 +132,130 @@ func GetComponentStateFromSpec(
...
+
+func getReplicaSummariesForPods(jobPods []corev1.Pod) []deploymentModels.ReplicaSummary {
+	var replicaSummaries []deploymentModels.ReplicaSummary
+	for _, pod := range jobPods {
+		replicaSummaries = append(replicaSummaries, deploymentModels.GetReplicaSummary(pod))
+	}
+	return replicaSummaries
+}
+
+func slicePodNamesAndPodsFromMap(podMap *map[string]corev1.Pod) (*[]string, *[]corev1.Pod) {
```
Comment (nilsgstrabo) [inline api/deployments/component_handler.go:212]: I don't think it is neccessary to use pointers to slices. A slice is a struct with a pointer to an array.

Also, maps are always reference types. No need to use pointers to map as parameters.

```
func changeMap(m map[string]string) {
	m["b"] = "bb"
}

func main() {
	var m map[string]string = make(map[string]string)
	m["a"] = "aa"
	changeMap(m)
	fmt.Println(m)
	// prints map[a:aa b:bb]
}
```

**Review by nilsgstrabo — APPROVED** [review]

File: api/deployments/deployment_controller.go (hunk 1 above)
Comment (satr) [reply api/deployments/deployment_controller.go]: fixed

File: docs/docs.go (hunk 3 above)
Comment (satr) [reply docs/docs.go]: copy-paste. fixed :)

File: api/deployments/component_handler.go (hunk 4 above)
Comment (satr) [reply api/deployments/component_handler.go]: fixed

**Pull request merged**